extras/ioc_hash_scan.sh

```diff
@@ -0,0 +1,189 @@
+# export AWS_REGION="us-east-1" >> "$HOME/.bashrc"
+# Then reload your shell, like
+# source ~/.bashrc
+#
+# This script assumes that it exists in the ioc-scanner/extras/ directory.
+# If it does not, please edit the variable $pydir to point to
+# the directory containing ioc_scanner.py
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Directory path of the ioc_scanner.py file
+pydir="../src/ioc_scan/"
+
+if [ $# -eq 0 ] || [ "$1" = "-h" ] || [ "$1" = "--help" ] || [ $# -lt 2 ]; then
+  echo Usage: "$0" instance-list-file \<AWS_PROFILE\>
+  exit 1
+fi
+
+# Check if instance list file exists.
+if [ ! -f "$1" ]; then
+  echo Instance List file "$1" does not exist - exiting.
+  exit 1
+fi
+
+# Read instance id strings from file.  [[ -n "$line" ]] handles the case where
+# the last line doesn't end with a newline.
+serverList=()
+while IFS= read -r line || [[ -n "$line" ]]; do
+  serverList+=("$line")
+done < "$1"
+
+if [ ${#serverList[@]} -eq 0 ]; then
+  echo No Instances found in "$1" - exiting.
+  exit 1
+fi
+
+# Path of aws credentials file
+AWSPROF="$2"
+
+today=$(date +%Y-%m-%d)
+logfile="$HOME/$today-ioc-scanner-hashscan.log"
+
+# Suppress some verbose stdout.
+# Suppress stderr of pkill command
+exec > >(grep --invert-match 'Starting\|Exiting')
+exec 2> >(grep --invert-match 'SIGTERM')
+
+## FUNCTIONS
+function getOSType() {
+  OSNAME=$(AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command=cat /etc/os-release | grep --extended-regexp ^NAME=" \
+    --output text)
+  echo "$OSNAME" | grep NAME | cut -d'"' -f2 | cut -d' ' -f1
+}
+
+function getHost() {
+  HOST=$(AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command=hostname" \
+    --output text)
+  echo "$HOST" | grep --invert-match "session" | sed '/^$/d'
+}
+
+function portForward() {
+  AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartPortForwardingSession --parameters="localPortNumber=5555,portNumber=6666"
+}
+
+function installNC() {
+  if [[ "$OS" == "Debian" ]]; then
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo apt-get --yes install netcat'"
+  else
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo dnf --assumeyes install netcat'"
+  fi
+}
```

Comment:
If you install the newer `nmap-ncat` then you can use the long-form CLI options.
```suggestion
function installNC() {
  if [[ "$OS" == "Debian" ]]; then
    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
      aws --profile="$AWSPROF" --region="$AWS_REGION" \
      ssm start-session --target="$i" \
      --document=AWS-StartInteractiveCommand \
      --parameters="command='sudo apt-get --yes install nmap-ncat'"
  else
    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
      aws --profile="$AWSPROF" --region="$AWS_REGION" \
      ssm start-session --target="$i" \
      --document=AWS-StartInteractiveCommand \
      --parameters="command='sudo dnf --assumeyes install nmap-ncat'"
  fi
}
```

Reply:
Are long-form options that critical? I know you like them, but do we have to install a different package just so that we can use a long-form option? I'm confused on the deeper requirement here.

Reply:
It's just a suggestion.  I was wondering why I had long-form options on my Linux machine and you didn't, then I realized that was because you are installing a different `netcat` package.

Reply:
I would say that unless there are benefits beyond long-form options, I would say use the standard `netcat`.